extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTools.java

```diff
@@ -603,8 +603,12 @@ private static void writeTablesImpl(
                         final long numBytes = ParquetTableWriter.write(source, definition, writeInstructions,
                                 tableDestination, outputStream, Collections.emptyMap(),
                                 (List<ParquetTableWriter.IndexWritingInfo>) null, metadataFileWriter, computedCache);
-                        writeInstructions.onWriteCompleted().ifPresent(callback -> callback.onWriteCompleted(
-                                tableDestination, source.size(), numBytes));
+                        writeInstructions.onWriteCompleted()
+                                .ifPresent(callback -> callback.onWriteCompleted(CompletedParquetWrite.builder()
+                                        .destination(tableDestination)
+                                        .numRows(source.size())
```

Comment:
I wonder if it's more appropriate to plumb the callback down through `ParquetTableWriter.write`? In that way, `write` doesn't need to return the number of bytes anymore, and it's going to be responsible internally for calling the number of rows (instead of making the caller do `source.size()`; I do see the safety check is at the inner layer calling `checkInitiateSerialTableOperation`).